Elaborate on test IDs in the test IDs guide: Naming best practices (#4487)
wix/Detox: docs/guide/test-id.mdx

import Android from './partials/_find-test-id-android.mdx';
import IOS from './partials/_find-test-id-ios.mdx';

# Adding test ID's to your components

:::info Note

> This guide is optimized for React Native apps, but can be generalized for testing any app, including native apps.

Reviewer: ```suggestion
This guide is written for React Native apps, but can be generalized for testing any app, including native apps.
```

:::

> While [view-element matching](../api/matchers.md) can be done in numerous ways, it is always the best idea to match based on something unique and decoupled, as it ensures that the test code would be clear, stable and sustainable over time. We recommend assigning unique test ID's to the elements you're aiming to interact with in your tests, and prefering matching based on those rather than on anything else. Test ID's are the least likely to change over time (compared with raw text, for example), and are locale-agnostic.

Reviewer: IMO there's another point for test-ids:

```suggestion
While [view-element matching](../api/matchers.md) can be done in numerous ways, it is always the best idea to match based on something unique and decoupled, as it ensures that the test code would be clear, stable and sustainable over time. 

We recommend assigning unique test ID's to the elements you're aiming to interact with in your tests, and preferring matching based on those rather than on anything else. Test ID's are the least likely to change over time (compared with raw text, for example), and are locale-agnostic. Additionally, utilizing unique test IDs across the app not only simplifies the identification and interaction with specific elements but also enhances code navigability, making it easier to locate elements when traversing the codebase.
```

That way, you could refer to specific elements in Detox tests via the most basic and least ambiguous `by.id` matchers, e.g.:

```js

> expect(element(by.id('YourCustomComponent'))).toBeVisible(); // the element is visible
> expect(element(by.id('YourCustomComponent.label'))).toHaveText('Some text'); // the label has some text

Reviewer: explicit enough I think
```suggestion
expect(element(by.id('YourCustomComponent'))).toBeVisible();
expect(element(by.id('YourCustomComponent.label'))).toHaveText('Some text');
```

Author: lol... I agree
